frontend/src/app/accountCreation/SecurityQuestion/SecurityQuestion.tsx

```diff
@@ -10,34 +11,72 @@ import {
   accountCreationSteps,
   securityQuestions,
 } from "../../../config/constants";
+import { AccountCreationApi } from "../AccountCreationApiService";
 
 export const SecurityQuestion = () => {
+  // State setup
   const [securityQuestion, setSecurityQuestion] = useState("");
   const [securityQuestionError, setSecurityQuestionError] = useState("");
   const [securityAnswer, setSecurityAnswer] = useState("");
   const [securityAnswerError, setSecurityAnswerError] = useState("");
+  const [loading, setLoading] = useState(false);
+  const [submitted, setSubmitted] = useState(false);
 
-  const validateSecurityQuestion = () => {
+  const validateSecurityQuestion = (): boolean => {
+    let error = "";
     if (securityQuestion === "") {
-      setSecurityQuestionError("Enter a security question");
-    } else {
-      setSecurityQuestionError("");
+      error = "Enter a security question";
     }
+    setSecurityQuestionError(error);
+    return error === "";
   };
 
-  const validateSecurityAnswer = () => {
+  const validateSecurityAnswer = (): boolean => {
+    let error = "";
     if (securityAnswer === "") {
-      setSecurityAnswerError("Enter a security answer");
-    } else {
-      setSecurityAnswerError("");
+      error = "Enter a security answer";
     }
+    setSecurityAnswerError(error);
+    return error === "";
   };
 
-  const handleSubmit = () => {
-    validateSecurityQuestion();
-    validateSecurityAnswer();
+  const handleSubmit = async () => {
+    if (validateSecurityQuestion() && validateSecurityAnswer()) {
+      setLoading(true);
+      try {
+        await AccountCreationApi.setRecoveryQuestion(
+          securityQuestion,
+          securityAnswer
+        );
+      } catch (error) {
+        console.error(error);
```

Comment:
Probably want a more user-friendly way to handle this! Right now it'll just fail, log to the console, and then redirect to `mfa-select` and _appear_ like everything is fine

Reply:
Fixed in 3f99d71